spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -135,6 +135,15 @@ def get_parser():
         action="help",
         help="Show this help message and exit."
     )
+    optional.add_argument(
+        '-s-template-id',
+        metavar=Metavar.int,
+        type=int,
+        help="Segmentation from the template to associate with the flag '-s'. Typically, the spinal cord segmentation "
+             "is used, but if it is available, the white matter mask would produced better registration results. The ID"
+             "is indicated in the file 'template/info_label.txt' of the template indicated by the flag '-t'.",
+        default=3
```

Comment:
I think this is probably going to drive people crazy. It's a double-layer of indirection, via an integer. Could it at least be an enum instead of an int?

But, looking at the only template people actually use, `data/PAM50/template/` (to remind myself and the future: this only shows up after `./install_sct`; it's source is https://github.com/sct-data/PAM50/releases/download/r20201104/PAM50-r20201104.zip), the options are:


```
# Template labels
# Keyword=IndivLabels (Please DO NOT change this line)
# ID, name, file
0, T1-weighted template, PAM50_t1.nii.gz
1, T2-weighted template, PAM50_t2.nii.gz
2, T2S-weighted template, PAM50_t2s.nii.gz
3, spinal cord mask (binary), PAM50_cord.nii.gz
4, white matter mask (probabilistic), PAM50_wm.nii.gz
5, gray matter mask (probabilistic), PAM50_gm.nii.gz
6, cerebrospinal fluid mask (binary), PAM50_csf.nii.gz
7, spinal cord mask with discrete vertebral levels, PAM50_levels.nii.gz
8, spinal cord mask with continuous vertebral levels, PAM50_levels_continuous.nii.gz
9, point-wise vertebral body labels, PAM50_label_body.nii.gz
10, point-wise intervertebral disc labels, PAM50_label_disc.nii.gz
11, point-wise posterior intervertebral disc labels, PAM50_label_discPosterior.nii.gz
12, spine mask (binary), PAM50_spine.nii.gz
13, spinal cord centerline, PAM50_centerline.nii.gz
14, point-wise spinal level labels, PAM50_label_spinal_levels.nii.gz
```

So, `info_label.txt` is a manifest, or maybe a dictionary. And it already has unique IDs. So instead of taking an int here, what if you took the string "cord", or "wm" or "gm". That way the scripts people run and what they see on disk match up more directly. You can keep `info_label.txt` for the full names, but for the future I'd also think about dropping its first column and swapping the third and second around so it looks like like the key-value dictionary it is.

You'll have to rework (or maybe just drop?) `get_file_label()`, then.

As written, especially with this explanation, it looks like *every* template has the same set of options. But that's not true, is it? If that was true, why ship `info_label.txt` with each?

Reply:
@kousu these are excellent thoughts-- quick context (no time to elaborate more): a few years ago we did use more intuitive naming (sc, wm, etc.) however for reasons i vaguely remember (we need to dig in the past PRs, i don't have time to do that now) this caused some issues because as we were prototyping register to template functionality and adding more contrast (t1, t2s, etc.) it because cumbersome to use naming (eg: there would be a "label" for "vertebral label" and then we introduced spinal label, so "label" was no more specific enough, etc.), so we ended up using numerical IDs. However, i've been unhappy about this choice for the past few years-- i'd like to come back to your suggestion-- if you or someone could please open an issue to get this started that would be great-- for now, this PR needs go get in because users are waiting after us-- they have urgent analyses to perform-- thanks

Reply:
#3303 

I also edited your explanation so that it doesn't talk about white matter, because each template has many or few segmentations and there's no reason any of them have to be white matter or that the user would want to use that one. Unless this is somewhere where my knowledge is failing me.

Reply:
thank you @kousu, actually i think i will put back WM, because in 99% of use cases, ppl have the cord seg, in 20% of the cases they have cord and WM, and in <1% of the cases they have another seg (eg: MS lesion, spine seg, etc.).

Reply:
thank you @kousu, actually i think i will put back WM, because in 99% of use cases, ppl have the cord seg, in 20% of the cases they have cord and WM, and in <1% of the cases they have another seg (eg: MS lesion, spine seg, etc.).